### PR TITLE
Add newBuilder overload for resolving directly to a builder.

### LINKED ISSUE #2063: Ability to either call resolve() on a HttpUrl.Builder -or- an overload of resolve() that returns an HttpUrl.Builder

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #2063.

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/HttpUrl.java [hunk 1]
```diff
@@ -620,6 +622,17 @@ public Builder newBuilder() {
   }
 
   /**
+   * Returns a builder for the URL that would be retrieved by following {@code link} from this URL,
+   * allowing it to be manipulated further. Equivalent and more efficient than calling
+   * {@link #resolve(String) resolve(link).newBuilder()}.
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/HttpUrl.java]: Not sure I love the circular definition!

File: okhttp/src/main/java/okhttp3/HttpUrl.java [hunk 2]
```diff
@@ -597,11 +597,13 @@ public String fragment() {
     return fragment;
   }
 
-  /** Returns the URL that would be retrieved by following {@code link} from this URL. */
+  /**
+   * Returns the URL that would be retrieved by following {@code link} from this URL. Equivalent
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/HttpUrl.java]: It was missing before, but maybe mention that this returns `null` if the link cannot be resolved?

**Comment by swankjesse** [comment]
LGTM

File: okhttp/src/main/java/okhttp3/HttpUrl.java (hunk 1 above)
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/HttpUrl.java]: Removed. Wasn't really attached to it.

File: okhttp/src/main/java/okhttp3/HttpUrl.java (hunk 2 above)
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/HttpUrl.java]: Added to both `resolve` and `newBuilder` with the same wording as `parse`.

**Pull request merged**